File: server/models/profile/login_user.js
export default function buildLoginUser() {
  return function loginUser({ username, password }) {
    if (!username) {
      throw new Error("username_empty");
    }

    if (!password) {
      throw new Error("password_empty");
    }

    return Object.freeze({
      username,
      password,
    });
  };
}

File: lib/backend/Models/authentication/loginUser.dart
class LoginUser {
  String name;
  String pass;

  LoginUser({String username, String password}) {
    this.name = username;
    this.pass = password;
  }
  
  String get username{
    return this.name;
  }

  String get password{
    return this.pass;
  }

  set password(String password) {
    this.pass=password;
  }

}
